src/components/device/ha-device-trigger-picker.ts

```diff
@@ -0,0 +1,155 @@
+  platform: string;
+  domain: string;
+  device_id: string;
+  entity_id?: string;
+  type: string;
+}
+
+const fetchDeviceTriggers = (hass, deviceId) =>
+  hass.callWS<DeviceTrigger[]>({
+    type: "device_automation/list_triggers",
+    device_id: deviceId,
+  });
+
+function isObject(v) {
+  return "[object Object]" === Object.prototype.toString.call(v);
+}
+
+JSON.sort = function(o) {
+  if (Array.isArray(o)) {
+    return o.sort().map(JSON.sort);
+  } else if (isObject(o)) {
+    return Object.keys(o)
+      .sort()
+      .reduce(function(a, k) {
+        a[k] = JSON.sort(o[k]);
+
+        return a;
+      }, {});
+  }
+
+  return o;
+};
+
+class HaEntityPicker extends LitElement {
+  public hass?: HomeAssistant;
+  public localize?: LocalizeFunc;
+  @property() public label?: string;
+  @property() public value?: string;
+  @property() public deviceId?: string;
+  @property() public triggers?: DeviceTrigger[];
+
+  private _sortedTriggers = memoizeOne((triggers?: DeviceTrigger[]) => {
+    return triggers || [];
+  });
+
+  protected render(): TemplateResult | void {
+    return html`
+      <paper-dropdown-menu-light .label=${this.label}>
+        <paper-listbox
+          slot="dropdown-content"
+          .selected=${this._value}
+          attr-for-selected="data-trigger"
+          @iron-select=${this._triggerChanged}
+        >
+          <paper-item
+            data-trigger=${JSON.stringify({
+              device_id: this.deviceId,
+              platform: "device",
+            })}
+          >
+            No trigger
+          </paper-item>
+          ${this._sortedTriggers(this.triggers).map(
+            (trigger) => html`
+              <paper-item data-trigger=${JSON.stringify(JSON.sort(trigger))}>
+                ${this.localize(
+                  `ui.panel.config.automation.editor.triggers.type.device.trigger_type.${
+                    trigger.type
+                  }`,
+                  "entity_id",
+                  trigger.entity_id
+                    ? computeStateName(this.hass.states[trigger.entity_id])
+                    : "",
+                  "event",
+                  trigger.event
+                )}
+              </paper-item>
+            `
+          )}
+        </paper-listbox>
+      </paper-dropdown-menu-light>
+    `;
+  }
+
+  private get _value() {
+    return this.value || "";
+  }
+
+  protected updated(oldProps) {
+    console.log("_triggerChanged");
+    if (oldProps.has("deviceId") && oldProps.get("deviceId") != this.deviceId) {
+      if (this.deviceId) {
+        fetchDeviceTriggers(this.hass!, this.deviceId).then((trigger) => {
+          this.triggers = trigger.triggers;
+        });
+      } else {
+        this.triggers = [];
+      }
+    }
+  }
+
+  private _triggerChanged(ev) {
+    console.log("_triggerChanged");
+    const tmp = JSON.parse(ev.detail.item.dataset.trigger);
+    const newValue = ev.detail.item.dataset.trigger;
+    if (newValue !== this._value) {
```

Comment:
this will always be true because objects are not the same .

Reply:
Both are strings, so the comparison works.
On the other hand is this a reasonable way..
https://github.com/home-assistant/home-assistant-polymer/blob/26a29db23b1ecea8bd8cb25499d48c098640f52b/src/components/device/ha-device-trigger-picker.ts#L130-L133

..to get this code called:
https://github.com/home-assistant/home-assistant-polymer/blob/26a29db23b1ecea8bd8cb25499d48c098640f52b/src/panels/config/js/trigger/device.js#L45-L47

Or is there a better way to listen to the dropdown from device.js?